cmd/operator/main.go

```diff
@@ -87,6 +87,31 @@ func main() {
 
 	version.Print()
 
+	// Get the watched namespace. This is originally sourced from the OperatorGroup associated with the CSV.
+	// Because the WMCO CSV only supports the OwnNamespace InstallMode, the watch namespace will always be the namespace
+	// that WMCO is deployed in.
+	watchNamespace, err := getWatchNamespace()
+	if err != nil {
+		setupLog.Error(err, "failed to get watch namespace")
+		os.Exit(1)
+	}
+	// WMCO can only function properly in this namespace, as OLM is unable to adjust the WICD RoleBinding to target a
+	// ServiceAccount in any namespace but this one.
+	supportedNamespace := "openshift-windows-machine-config-operator"
+	if watchNamespace != supportedNamespace {
+		setupLog.Error(err, "the operator must be deployed in the namespace "+supportedNamespace)
+	}
+	// This is a defensive check to ensure that the WMCO CSV was not changed from only supporting OwnNamespace only.
+	// Check that the OperatorGroup + CSV were not deployed with a cluster scope (namespace = ""), and that the
+	// OperatorGroup does not target multiple namespaces. This should not be able to happen as both `AllNamespaces` and
+	// `MultiNamespaces` are not supported InstallModes.
+	if watchNamespace == "" || strings.Contains(watchNamespace, ",") {
+		setupLog.Error(err, "WMCO has an invalid target namespace. "+
+			"OperatorGroup target namespace must be a single, non-cluster-scoped value", "target namespace",
+			watchNamespace)
+		os.Exit(1)
+	}
+
```

Comment:
Please wrap this into a ` namespaceCheck()` helper. This `main()` function is growing.